### PR TITLE
feat(StructureList): refactor class components, add aria table roles

### PR DESCRIPTION
**Pull request by dakahn**
part of #1937 

We're working on #1937 and taking the opportunity to update StructuredList from an es6 class components to functional components w/hooks. StructuredList also seemed to to have no table markup whatsoever so i've added some basic aria roles making the component identifiable as a table to screen readers. 

#### Testing / Reviewing

This PR shouldn't have any visible changes or functionality changes whatsoever.

### CODE REVIEW

**Comment by dakahn** [comment]
@tw15egan can't figure out why the unit test for unique IDs is failing. `getInstanceId()` seems to be working as intended -- at least I think it is 😕 

Don't have a whole lot of enzyme experience but is that failing test _looking for a `uid` prop_?

File: packages/react/src/components/StructuredList/StructuredList.js [hunk 1]
```diff
@@ -5,321 +5,309 @@
...
+   * Specify an optional className to be applied to the node
+   */
+  className: PropTypes.string,
+};
+
+export function StructuredListBody(props) {
+  const { children, className, ...other } = props;
+  const classes = classNames(`${prefix}--structured-list-tbody`, className);
+  const { labelRows, setLabelRows } = useState(null);
+  const { rowSelected, setRowSelected } = useState(0);
```
Comment (dakahn) [inline packages/react/src/components/StructuredList/StructuredList.js]: this is kinda weird, probably something i'm missing, but `setLabel` and `rowSelected` were tracked in state, but unused?

Comment (tw15egan) [reply packages/react/src/components/StructuredList/StructuredList.js]: AFAIK `StructuredList` hasn't been touched in quite some time and could definitely have some leftover unused hooks that may need to be refactored

**Comment by andreancardona** [comment]
@dakahn I think we need to update tests!

File: packages/react/src/components/StructuredList/StructuredList.js (hunk 1 above)
Comment (dakahn) [reply packages/react/src/components/StructuredList/StructuredList.js]: Totally the case -- just got rid of em

**Comment by dakahn** [comment]
`wrapper.instance()` checks against a class instance for a prop called `uid`. I updated the test to instead just check for unique id attributes being assigned correclty 👍🏽 @tw15egan

**Review by andreancardona — APPROVED** [review]
yay!

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Comment by tw15egan** [comment]
@dakahn just confirming, is selection via keyboard the next step as part of #1937?

**Comment by dakahn** [comment]
@tw15egan yep! This refactor is preliminary work _before_ we start hacking on 1937 👍🏽

**Pull request merged**